volatility3/framework/plugins/linux/kmsg.py

```diff
@@ -0,0 +1,387 @@
+        "warn",      # warning conditions
+        "notice",    # normal but significant condition
+        "info",      # informational
+        "debug",     # debug-level messages
+    )
+
+    FACILITIES = (
+        "kern",      # kernel messages
+        "user",      # random user-level messages
+        "mail",      # mail system
+        "daemon",    # system daemons
+        "auth",      # security/authorization messages
+        "syslog",    # messages generated internally by syslogd
+        "lpr",       # line printer subsystem
+        "news",      # network news subsystem
+        "uucp",      # UUCP subsystem
+        "cron",      # clock daemon
+        "authpriv",  # security/authorization messages (private)
+        "ftp"        # FTP daemon
+    )
+
+    def __init__(
+        self,
+        context: interfaces.context.ContextInterface,
+        config: interfaces.configuration.HierarchicalDict
+    ):
+        self._context = context
+        self._config = config
+        self.layer_name = self._config['primary']  # type: ignore
+        symbol_table_name = self._config['vmlinux']  # type: ignore
+        self.vmlinux = contexts.Module(context, symbol_table_name, self.layer_name, 0)  # type: ignore
+        self.long_unsigned_int_size = self.vmlinux.get_type('long unsigned int').size
+
+    @classmethod
+    def run_all(
+            cls,
+            context: interfaces.context.ContextInterface,
+            config: interfaces.configuration.HierarchicalDict
+    ) -> Iterator[Tuple[str, str, str, str, str]]:
+        """It calls each subclass symtab_checks() to test the required
+        conditions to that specific kernel implementation.
+
+        Args:
+            context: The volatility3 context on which to operate
+            config: Core configuration
+
+        Yields:
+            kmsg records
+        """
+
+        symbol_table_name = config['vmlinux']  # type: ignore
+        layer_name = config['primary']  # type: ignore
+        vmlinux = contexts.Module(context, symbol_table_name, layer_name, 0)  # type: ignore
+
+        kmsg_inst = None  # type: ignore
+        for subclass in class_subclasses(cls):
+            if not subclass.symtab_checks(vmlinux=vmlinux):
+                vollog.log(constants.LOGLEVEL_VVVV,
+                           "Kmsg implementation '%s' doesn't match this memory dump", subclass.__name__)
+                continue
+
+            vollog.log(constants.LOGLEVEL_VVVV, "Kmsg implementation '%s' matches!", subclass.__name__)
+            kmsg_inst = subclass(context=context, config=config)
+            # More than one class could be executed for an specific kernel
+            # version i.e. Netfilter Ingress hooks
+            # We expect just one implementation to be executed for an specific kernel
+            yield from kmsg_inst.run()
+            break
+
+        if kmsg_inst is None:
+            vollog.error("Unsupported Netfilter kernel implementation")
+
+    @abstractmethod
```

Comment:
I'm a little confused by the need for the abstractmethod stuff?  Seems like a bunch of ABC stuff coming in here for a single method (when there are other shared methods, like `symbtab_checks`, even though they're classmethods)?  It seems they're class methods just so that a few variables can be stashed in the instance rather than passing them as parameters?  I don't particularly have a problem with it, I just wonder which is easier to read?  Happy with this if you prefer it this way...

Reply:
re abstract methods: I make sure the subclasses will implement _run_ and _symtab_checks_.  
Note that I forgot to make _symtab_checks_ an _abstractmethod_ but it was just fixed. I also added a docstring to clarify the _symtab_checks_ usage.